overlay/agent.cpp

```diff
@@ -207,12 +167,70 @@ Try<Owned<ManagerProcess>> ManagerProcess::create(
     }
   }
 
+  Try<Owned<MasterDetector>> detector =
```

Comment:
Can we do this?
```
auto getDetector = [=]()->Try<Owned<MasterDetector>> {
  if (_detector.isSome()) {
        return _detector.get();
  }

  return ManagerProcess::createDetector(agentConfig);
}

Try<Owned<MasterDetector>> detector = getDetector();
if (detector.isError()) {
  return ....
}

Reply:
Thank you for pointing it out! I've removed the second arguments from `createDetector` function and re-write this code.

https://github.com/dcos/dcos-mesos-modules/blob/6649d98e5ee58b20e159f81b967e622f6acb6175/overlay/agent.cpp#L170-L175